test/runtests.jl

```diff
@@ -55,6 +55,8 @@ end
     @test AutoViz._rendermode == :fancy
 
     @test_deprecated render([roadway, veh1, veh2, veh3])
+    # render(Frame([veh1, veh2, veh3]), camera_zoom=15., camera_center=VecE2(1.,1.), camera_motion=TargetFollowCamera(target_id=1))  # TODO: multiple dispatch not working on update_camera!
```

Comment:
`update_camera!` does not get resolved as expected by multiple dispatch. Ideally I would like to enforce that `I` is the same in `TargetFollowCamera{I}` and `Frame{Entity{S,D,I}}` but this doesn't seem to work. The exact reason is beyond me. Could you have a look @MaximeBouton ?
To reproduce, just run the tests locally and uncomment the above line with `TargetFollowCamera`

Reply:
It is a pretty sneaky issue, I am not sure I fully understand, but I think it is related to the fact that `Vector{Int64}` is not a subtype of `Vector{Real}`. 

I hope this example helps: 

```julia 
julia> scene = Frame([veh1, veh2, veh3]) # this is what gets passed in to the `update_camera` method, note the location of the curly braces, the type of scene is Frame{something abstract}
Frame{Entity{VehicleState,D,Int64} where D}(3 entities)

julia> typeof(scene) <: Frame
true

julia> typeof(scene) <: Frame{Entity{S,D,I}} where {S,D,I} # this is false, hence the method error, let's investigate
false

julia> Frame{Entity{VehicleState,VehicleDef,Int64}} <: Frame{Entity{S,D,I}} where {S,D,I}
true

julia> Frame{Entity{VehicleState,VehicleDef,Real}} <: Frame{Entity{S,D,I}} where {S,D,I}
true

julia> (Frame{Entity{VehicleState,VehicleDef,R}} where R) <: Frame{Entity{S,D,I}} where {S,D,I} 
true

julia> (Frame{Entity{VehicleState,VehicleDef,R} where R}) <: Frame{Entity{S,D,I}} where {S,D,I}
false
```

Reply:
Good catch! I didn't realize the curly braces were in different positions, that must be the reason!!

Reply:
Ok so the following works:
```
function update_camera!(rendermodel::RenderModel, camera::TargetFollowCamera{I}, scene::Frame{Entity{S,D,I} where D}) where {S,I}
```
I think it's related to the fact that vehicle definition is different in some of these cases, so `Frame{Entity{S,D,I}} where {S,D,I}` enforces all entities to be the same, while `Frame{Entity{S,D} where {S,D}} where I` enforces only the `id` to be consistent while `state` and `def` can vary among entities in the frame... Food for thought, I need to digest this :-)